Engine/media/audio/audio_system.h

```diff
@@ -0,0 +1,16 @@
+#ifndef __AC_MEDIA_AUDIO_SYSTEM_H
+#define __AC_MEDIA_AUDIO_SYSTEM_H
+
+#include "media/audio/audiodefines.h"
+#include "media/audio/ambientsound.h"
+
+#include "media/audio/audio.h"
+
+#include "media/audio/soundcache.h"
+
+#include "media/audio/soundclip.h"
+#include "media/audio/sound.h"
+
+#include "media/audio/queuedaudioitem.h"
+
```

Comment:
Uh! I was specifically splitting them up to avoid including something that does not need to be in every particular unit. :/

There were couple of more big headers in the project like this which I wanted to remove. Now I will be wanting to remove this. Or refactor whole thing again to prevent the need of such include list.

Reply:
Do we need such fine grained include files though? What are we gaining? I'm not suggesting one giant include file, but perhaps one per system would be managable? As it is, I had to add/remove include files just for audio as I added/moved code around.

Reply:
Maybe not if few headers are always included together into source files.

But I think it helps to understand the ties between program parts, as well as the problems with these ties. Right now  audio code has different layers of abstraction mashed together: from script commands to clip interface and sound cache (which should be completely internal perhaps).

Reply:
When I am saying "helps to understand", I mean having this (small) inconvenience is a kind of reminder to self that something is wrong with it.

Reply:
I was thinking maybe we only need to worry about specific headers within the media/audio directory. Source outside of the directory could use the audio_system.h directly. When we tease out higher and lower level apis, we could split the audio directory and have separate headers.

Reply:
Hmm ok, this may have sense too.

Reply:
I'd propose to only include this in *.cpp files though.

Reply:
> I'd propose to only include this in *.cpp files though.
Agreed. In this pull request I deliberately didn't add audio_system.h to any headers. We should still aim to only forward declare types if possible.

Reply:
> But I think it helps to understand the ties between program parts, as well as the problems with these ties.

it also helps to speed up the build